sdk/python/lib/pulumi/automation/_cmd.py

```diff
@@ -35,46 +42,259 @@ def __str__(self) -> str:
+        version: Optional[VersionInfo] = None,
+        skip_version_check: bool = False,
+    ) -> PulumiCommand:
+        """
+        Installs the Pulumi CLI. If root is not specified, the CLI is
+        installed to ~/.pulumi/versions/<version>. If version is not
+        specified, the version matching the SDK version is installed.
+
+        :param root: The root directory to install the CLI to.
+        :param version: The version of the CLI to install.
+        :skip_version_check: If true, the version validation will be skipped.
+        """
+        if not version:
+            version = sdk_version
+        if not root:
+            root = os.path.join(
+                os.path.expanduser("~"), ".pulumi", "versions", str(version)
+            )
+
+        try:
+            return PulumiCommand(
+                root=root, version=version, skip_version_check=skip_version_check
+            )
+        except Exception:
+            pass  # Ignore
+
+        if os.name == "nt":
+            cls._install_windows(root, version)
+        else:
+            cls._install_posix(root, version)
+
+        return PulumiCommand(
+            root=root, version=version, skip_version_check=skip_version_check
+        )
+
+    @classmethod
+    def _install_windows(cls, root: str, version: VersionInfo):
+        # TODO: Once we're on python 3.12 we can use a `with` context manager with `delete_on_close=True` and `delete=True` here
+        # pylint: disable-next=consider-using-with
+        script = tempfile.NamedTemporaryFile(delete=False, suffix=".ps1")
+        try:
+            _download_to_file("https://get.pulumi.com/install.ps1", script.name)
+            script.close()  # The file was opened for writing, so we need to close it before executing it.
+            command = "powershell.exe"
+            sys_root = os.getenv("SystemRoot")
+            if sys_root:
+                command = os.path.join(
+                    sys_root,
+                    "System32",
+                    "WindowsPowerShell",
+                    "v1.0",
+                    "powershell.exe",
+                )
+            subprocess.check_output(
+                [
+                    command,
+                    "-NoProfile",
+                    "-InputFormat",
+                    "None",
+                    "-ExecutionPolicy",
+                    "Bypass",
+                    "-File",
+                    script.name,
+                    "-NoEditPath",
+                    "-InstallRoot",
+                    root,
+                    "-Version",
+                    str(version),
+                ],
+                stderr=subprocess.STDOUT,
+            )
+        finally:
+            os.remove(script.name)
+
+    @classmethod
+    def _install_posix(cls, root: str, version: VersionInfo):
+        # TODO: Once we're on python 3.12 we can use a `with` context manager with `delete_on_close=True` and `delete=True` here
```

Comment:
```suggestion
        # TODO: Once we're on python 3.12 we can use a `with` context manager with `delete_on_close=False` and `delete=True` here
```

I think this change is still needed?